db/warehouse/migrate/20240118203430_add_scan_card_codes_table.rb

```diff
@@ -0,0 +1,12 @@
+class AddScanCardCodesTable < ActiveRecord::Migration[6.1]
+  def change
+    create_table(:hmis_scan_card_codes) do |t|
+      t.references :client, null: false
+      t.string :code, index: true, null: false # code to embed in scan card
```

Comment:
Oh good point, `value` is better.
I can add an expiration column but I'm not sure what that duration should be, would need to get some feedback on that

Reply:
@eanders what do you think about an expiration timestamp and duration?

Reply:
I think an optional expiration is a good idea, so we the card can be expired without having to be deleted.  I _think_ that would manifest as something that was set at creation time (and optionally updated later.)  I can also see having an indefinite card, depending on who is issuing it or what it is for.  I don't know why you'd expire a card that is just used to attach a meal to a person, but maybe there's a reason you'd want to check the PII of a person every year or two, so having the expiration could trigger a workflow in HMIS when scanned.

Reply:
Scan card expiration seems like a very cumbersome way to trigger that type of workflow.. at least with the current level of difficulty for re-creating and re-printing a scan card. 

This does feels like scope creep since it will change the interface t(o require user input for an expiration period), which I'm not sure we want. I'll add the column for now but hold off on UI changes until we can validate the utility of this feature.

Reply:
The expiration thing also seems like something that would site or installation-wide, not decided by the user every time they make a card. In that case maybe it would just be a job that deactivates (deletes) scan cards that are a certain number of years old. 

I added a commit with the optional expiration column though.